client/homepage/layout.js

```diff
@@ -34,43 +56,113 @@ const Layout = () => {
 		};
 	}, [] );
 
+	const { query, requestingTaskList, taskListHidden } = props;
+	const isTaskListEnabled = taskListHidden === false;
+	const isDashboardShown = ! isTaskListEnabled || ! query.task;
+
+	const renderColumns = () => {
+		return (
+			<Fragment>
+				{ showInbox && (
+					<div className="woocommerce-homepage-column is-inbox">
+						<div className="temp-content">
+							<Button
+								isPrimary
+								onClick={ () => {
+									setShowInbox( false );
+								} }
+							>
+								Dismiss All
+							</Button>
+						</div>
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+						<div className="temp-content" />
+					</div>
+				) }
+				<div
+					className="woocommerce-homepage-column"
+					ref={ content }
+					style={ {
+						position: isContentSticky ? 'sticky' : 'static',
+					} }
+				>
+					{ isTaskListEnabled && renderTaskList() }
+					<StatsOverview />
+				</div>
+			</Fragment>
+		);
+	};
+
+	const renderTaskList = () => {
+		if ( requestingTaskList ) {
+			return <TaskListPlaceholder />;
+		}
+
+		return (
+			<Suspense fallback={ <Spinner /> }>
+				<TaskList
+					query={ query }
+					inline
+				/>
+			</Suspense>
+		);
+	};
+
 	return (
 		<div
 			className={ classnames( 'woocommerce-homepage', {
 				hasInbox: showInbox,
 			} ) }
 		>
-			{ showInbox && (
-				<div className="woocommerce-homepage-column is-inbox">
-					<div className="temp-content">
-						<Button
-							isPrimary
-							onClick={ () => {
-								setShowInbox( false );
-							} }
-						>
-							Dismiss All
-						</Button>
-					</div>
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-					<div className="temp-content" />
-				</div>
-			) }
-			<div
-				className="woocommerce-homepage-column"
-				ref={ content }
-				style={ {
-					position: isContentSticky ? 'sticky' : 'static',
-				} }
-			>
-				<StatsOverview />
-			</div>
+			{ isDashboardShown
+				? renderColumns()
+				: isTaskListEnabled && renderTaskList()
+			}
 		</div>
 	);
 };
 
-export default Layout;
+Layout.propTypes = {
+	/**
+	 * If the task list option is being fetched.
+	 */
+	requestingTaskList: PropTypes.bool.isRequired,
+	/**
+	 * If the task list is hidden.
+	 */
+	taskListHidden: PropTypes.bool,
+	/**
+	 * Page query, used to determine the current task if any.
+	 */
+	query: PropTypes.object.isRequired,
+};
+
+export default compose(
+	withSelect( ( select ) => {
+		const {
+			getOptions,
```

Comment:
just an FYI that this is getting refactored in https://github.com/woocommerce/woocommerce-admin/pull/4144. If this is merged first, I'll update the code.